cmake: kobject: Fixed kobject text area overflow for optimization 0

AdithyaBaglody opened the pull request:
We need to allocate extra memory region for the hash function
that the gperf generates. The linker will need to maintain
the same location counter between multiple linker stages. Reserving memory
will help in maintaing correct location counters between multiple stages.
The required memory varies with optimization level.
In order to keep the size of the compiled object consistent,
the kobject_hash.c is always compiled with -Os.

fixes #5226 

Signed-off-by: Adithya Baglody <adithya.nagaraj.baglody@intel.com>

andrewboie commented on include/linker/kobject-text.ld:
```diff
@@ -1,8 +1,8 @@
-#define KOBJECT_TEXT_AREA	256
+#define KOBJECT_TEXT_AREA	512
 #else
-#define KOBJECT_TEXT_AREA	128
```
If this issue only affects the -O0 case, why are we increasing the size for the optimized case as well?

AdithyaBaglody commented:
Andrew I have made some of the changes. I am looking at extra_clfags to get the demarcation.
There is one other option which I feel might be cleaner. Force these generated files to always compile at a specific optimization level. what do you think Andrew?

andrewboie commented:
>  Force these generated files to always compile at a specific optimization level.

I like that a lot better, let's try this. We could then get rid of the if condition in the linker script.

AdithyaBaglody committed:
cmake: kobject: Fixed kobject text area overflow for optimization 0

We need to allocate extra memory region for the hash function
that the gperf generates. The linker will need to maintain
the same location counter between multiple linker stages. Reserving
memory will help in maintaing correct location counters between
multiple stages. The required memory varies with optimization level.
In order to keep the size of the compiled object consistent,
the kobject_hash.c is always compiled with -Os.

fixes #5226

Signed-off-by: Adithya Baglody <adithya.nagaraj.baglody@intel.com>

AdithyaBaglody commented:
delta:
updated the compiler flags for the required file.

andrewboie commented:
@AdithyaBaglody this is good if you can just fixup the commit message errors.

```
5: UC4 Line exceeds max length (74>72): "the same location counter between multiple linker stages. Reserving memory"
6: UC4 Line exceeds max length (73>72): "will help in maintaing correct location counters between multiple stages."
```
